src/utils/abi/index.ts

```diff
@@ -0,0 +1,25 @@
+import * as fs from "fs/promises";
+import { ContractInterface } from "ethers";
+import { isError } from "../../typeguards";
+
+/**
+ * @dev To retrieve a new ABI, the following is useful:
+ * CONTRACT_ADDRESS="0x1234..."
+ * URL="http://api.etherscan.io/api?module=contract&action=getabi&format=raw&address=${CONTRACT_ADDRESS}"
+ * curl -s "${URL}" | python -m json.tool > "./src/utils/abi/contracts/new-contract.json"
+ */
+
+/**
+ * @notice Retrieve an ABI desription from the set of known contracts.
+ * @param contractName Name of the contract ABI to retrieve.
+ * @returns Contract ABI as an ethers ContractInterface type.
+ */
+export async function getABI(contractName: string): Promise<ContractInterface> {
+  try {
+    return JSON.parse(await fs.readFile(`${__dirname}/contracts/${contractName}.json`, { encoding: "utf8" }));
```

Comment:
tbd: Will this run OK in serverless? I could imagine scenarios where we might want to point at the output code, rather than src...but maybe this is a non-issue?

Reply:
Turns out that it was missing entirely, so I've added it here: https://github.com/across-protocol/sdk-v2/pull/420/commits/2b6437b812485db12eb8c7e38f954edb576a6df1

Interested in your thoughts @james-a-morris. It feels a _little_ bit janky to me, but it works OK. Can it be improved?

Reply:
The one thing that I would recommend would be to cache these responses once they're read. `fs.readFile` is a heavy call.

Another note on further thought, this will probably fail in the FE. The reason being: `fs` is a node-specific directive and if we call this in via the FE it more than likely be undefined.

Reply:
> The one thing that I would recommend would be to cache these responses once they're read. `fs.readFile` is a heavy call.

Could we include a local caching mechanism to prevent having to resolve a specific ABI multiple times? File IO in node is heavy, and since we never change these ABIs, they're a good candidate for caching.
```suggestion
    return 
        CACHE[contractName] ??= JSON.parse(await fs.readFile(`${__dirname}/contracts/${contractName}.json`, { encoding: "utf8" }));
```